tests/watcher_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	client "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	TEST_GLOBAL_SVC_NAME = "echo-test-svc-global"
+	TEST_NS              = "default"
+	TEST_RETRIES         = 5
+)
+
+func GetEndpintSlices() {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	}
+
+	//use the current context in kubeconfig
+	config, err := client.BuildConfigFromFlags("", *kubeconfig)
+
+	fmt.Errorf(err.Error())
+	//Make sure config is not nil
+
+	client, err := kubernetes.NewForConfig(config)
+
+	//Make sure global-svc is created.
+	_, err = client.CoreV1().Services(TEST_NS).Get(context.Background(), TEST_GLOBAL_SVC_NAME, metav1.GetOptions{})
+
+}
+
+func TestGlobalService(t *testing.T) {
+	t.Logf("Testing if Service: %v is created", TEST_GLOBAL_SVC_NAME)
+
+	// Make request && make sure we get response from both cluster in X retries.
+	// Add retrieved responses inside map of string and make sure length is 2.
+	cluster := make(map[string]string)
+	tries := 0
+	for tries < TEST_RETRIES {
+		tries += 1
```

Comment:
What are we trying to do here? Why aren't we just using a normal for loop here, that's what the syntax sugar is for.

```suggestion
	for attempt := 0; attempt < MAX_ATTEMPTS; attempt++ {}
```

No need to use a `while` loop imo.

Reply:
+1 refactoring this